examples/tinywl/XdgStackWorkspace.qml

```diff
@@ -229,6 +302,38 @@ Item {
 
                     waylandSurface.setMaximize(false)
                 }
+
+                function onRequestMinimize() {
+                    if (waylandSurface.isResizeing)
+                        return
+
+                    if (waylandSurface.isMinimized)
+                        return
+
+                    focus = false;
+                    if (Helper.activeSurface === surface)
+                        Helper.activeSurface = null;
+
+                    visible = false;
+                    dockModel.append({ source: surface });
+                    waylandSurface.setMinimize(true)
+                }
+
+                function onRequestCancelMinimize() {
+                    if (waylandSurface.isResizeing)
+                        return
+
+                    if (!waylandSurface.isMinimized)
+                        return
+
+                    forceActiveFocus()
+                    Helper.activatedSurface = waylandSurface
+
+                    visible = true;
+
+                    dockModel.removeSurface(surface)
+                    waylandSurface.setMinimize(false)
+                }
```

Comment:
If the state has been maximized before being minimized, it should be restored to its maximum state.

Reply:
Record the old state